code-challenges.test.js

```diff
@@ -15,42 +15,141 @@
 
 // a) Create a test with expect statements using the variables provided.
 
-const secretCodeWord1 = "Lackadaisical"
-// Expected output: "L4ck4d41s1c4l"
-const secretCodeWord2 = "Gobbledygook"
-// Expected output: "G0bbl3dyg00k"
-const secretCodeWord3 = "Eccentric"
-// Expected output: "3cc3ntr1c"
+describe("codedMessage", () => {
+    test(" takes in a string and returns a coded message", () => {
+        const secretCodeWord1 = "Lackadaisical"
+//      Expected output: "L4ck4d41s1c4l"
+        const secretCodeWord2 = "Gobbledygook"
+//      Expected output: "G0bbl3dyg00k"
+        const secretCodeWord3 = "Eccentric"
+//      Expected output: "3cc3ntr1c"
+        expect(codedMessage(secretCodeWord1)).toEqual("L4ck4d41s1c4l")
+        expect(codedMessage(secretCodeWord2)).toEqual("G0bbl3dyg00k")
+        expect(codedMessage(secretCodeWord3)).toEqual("3cc3ntr1c")
+    })
+})
+
+/**   ReferenceError: codedMessage is not defined **/
+
+
+ // b) Create the function that makes the test pass.
+
+    /* Pseudo Code
+   * To start we will define our function giving it a name of codedMessage.
+   * The function will take in a string so, we will use str as a placeholder for the argument to be passed in.
+   * Our output will be the string returned, replacing characters with the criteria given: 'a' to 4, 'e' to 3, 'i' to 1, and 'o' to 0.
+   * To accomplish this we will first split the string into an array in order to iterate through each individual letter
+     using the higher order function .map.
+   * In tandem with a conditional if/elseif statement we make our comparisons and return statement.
+   * The replaced and returned values in the array are then rejoined to form the output string using .join
+   * Running the jests test the given values and return green if successful.
+*/
```

Comment:
Not bad pseudo code.  I would encourage you to try and use less words, short hand it a bit more.  But i can look at yours and know what your are trying to achieve 👍🏻